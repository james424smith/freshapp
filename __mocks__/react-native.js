import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
