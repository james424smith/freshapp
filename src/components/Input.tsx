import React, { useState, forwardRef, Ref } from "react";
import { Input } from "react-native-elements";
import { KeyboardTypeOptions, ReturnKeyTypeOptions } from "react-native";
import { Colors } from "react-native/Libraries/NewAppScreen";
import { useTheme } from "@react-navigation/native";
type AutoCapitalize = "none" | "sentences" | "words" | "characters";
interface Props {
  containerStyle: Record<string, any>;
  value: string;
  inputStyle: Record<string, any>;
  onChangeText: (arg: string) => void;
  placeholder: string;
  keyboardAppearance: "default" | "light" | "dark";
  autoFocus?: boolean;
  autoCapitalize: AutoCapitalize;
  autoCorrect: boolean;
  keyboardType: KeyboardTypeOptions;
  returnKeyType: ReturnKeyTypeOptions;
  blurOnSubmit: boolean;
  placeholderTextColor: string;
  onSubmitEditing?: () => void;
  rightIcon?: JSX.Element;
  inputContainerStyle: Record<string, any>;
  shouldUseFocusedStyles: boolean;
  secureTextEntry?: boolean;
  passwordInput?: Record<string, any>;
  testID?: string;
  disabled?: boolean;
}
const InputElement = forwardRef((props: Props, ref: Ref<Input>) => {
  const [isFocused, setIsFocused] = useState<boolean>(false);
  const theme = useTheme();
  const chooseContainerStyle = () => {
    if (props.shouldUseFocusedStyles) {
      return [
        props.containerStyle,
        {
          backgroundColor: isFocused
            ? Colors.white
            : theme.colors.colorInputText,
        },
      ];
    }
    return [props.containerStyle];
  };
  const chooseInputStyle = () => {
    if (props.shouldUseFocusedStyles) {
      return [
        props.inputStyle,
        {
          color: isFocused ? Colors.black : Colors.white,
        },
      ];
    }
    return [props.inputStyle];
  };
  return (
    <Input
      testID={props.testID}
      containerStyle={chooseContainerStyle()}
      onChangeText={props.onChangeText}
      value={props.value}
      secureTextEntry={props.secureTextEntry ?? false}
      inputStyle={chooseInputStyle()}
      keyboardAppearance={props.keyboardAppearance}
      placeholder={props.placeholder}
      autoFocus={props.autoFocus}
      autoCapitalize={props.autoCapitalize}
      autoCorrect={props.autoCorrect}
      keyboardType={props.keyboardType}
      returnKeyType={props.returnKeyType}
      onSubmitEditing={props.onSubmitEditing}
      rightIcon={props.rightIcon}
      onFocus={() => setIsFocused(true)}
      onBlur={() => setIsFocused(false)}
      blurOnSubmit={props.blurOnSubmit}
      placeholderTextColor={props.placeholderTextColor}
      inputContainerStyle={props.inputContainerStyle}
      ref={ref}
      disabled={props.disabled}
    />
  );
});
export default InputElement;
