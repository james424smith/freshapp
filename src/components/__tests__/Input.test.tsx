import React from "react";
import Input from "../Input";
import Colors from "../../constants/styles/colors";
import { render, cleanup, fireEvent } from "@testing-library/react-native";
import * as RNA from "react-native-appearance";

describe("BackIcon snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it("should render correct ui when scheme is dark", () => {
    Object.defineProperty(
      RNA,
      "useColorScheme",
      jest.fn().mockReturnValue("dark")
    );
    const { toJSON, getByTestId } = render(
      <Input
        testID={"input"}
        shouldUseFocusedStyles={true}
        containerStyle={{}}
        value="2345"
        inputStyle={{}}
        onChangeText={jest.fn()}
        placeholder="value"
        keyboardAppearance="default"
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={false}
        inputContainerStyle={{}}
        keyboardType="default"
        returnKeyType="default"
        placeholderTextColor={Colors.FAQInactiveColor}
      />
    );

    fireEvent(getByTestId("input"), "focus");
    fireEvent(getByTestId("input"), "blur");

    expect(toJSON()).toMatchSnapshot();
  });
  it("should render correct ui when scheme is dark when shouldUseFocusedStyles is false", () => {
    Object.defineProperty(
      RNA,
      "useColorScheme",
      jest.fn().mockReturnValue("dark")
    );
    const { toJSON } = render(
      <Input
        shouldUseFocusedStyles={false}
        containerStyle={{}}
        value="2345"
        inputStyle={{}}
        onChangeText={jest.fn()}
        placeholder="value"
        keyboardAppearance="default"
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={false}
        inputContainerStyle={{}}
        keyboardType="default"
        returnKeyType="default"
        placeholderTextColor={Colors.FAQInactiveColor}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
