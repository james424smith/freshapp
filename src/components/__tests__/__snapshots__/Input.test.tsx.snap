// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackIcon snapshot test should render correct ui when scheme is dark 1`] = `
<View
  style={
    Object {
      "backgroundColor": undefined,
      "paddingHorizontal": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#86939e",
        "flexDirection": "row",
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      blurOnSubmit={false}
      editable={true}
      keyboardAppearance="default"
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[MockFunction]}
      onFocus={[Function]}
      placeholder="value"
      placeholderTextColor="#8FAABF"
      rejectResponderTermination={true}
      returnKeyType="default"
      secureTextEntry={false}
      style={
        Object {
          "alignSelf": "center",
          "color": "#FFF",
          "flex": 1,
          "fontSize": 18,
          "minHeight": 40,
        }
      }
      testID="input"
      underlineColorAndroid="transparent"
      value="2345"
    />
  </View>
  <Text
    style={
      Object {
        "color": "#ff190c",
        "fontSize": 12,
        "margin": 5,
      }
    }
  />
</View>
`;

exports[`BackIcon snapshot test should render correct ui when scheme is dark when shouldUseFocusedStyles is false 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#86939e",
        "flexDirection": "row",
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      blurOnSubmit={false}
      editable={true}
      keyboardAppearance="default"
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[MockFunction]}
      onFocus={[Function]}
      placeholder="value"
      placeholderTextColor="#8FAABF"
      rejectResponderTermination={true}
      returnKeyType="default"
      secureTextEntry={false}
      style={
        Object {
          "alignSelf": "center",
          "color": "#242424",
          "flex": 1,
          "fontSize": 18,
          "minHeight": 40,
        }
      }
      underlineColorAndroid="transparent"
      value="2345"
    />
  </View>
  <Text
    style={
      Object {
        "color": "#ff190c",
        "fontSize": 12,
        "margin": 5,
      }
    }
  />
</View>
`;
